src/bindgen/ir/constant.rs

```diff
@@ -209,7 +210,19 @@ impl Literal {
                     fields: field_pairs,
                 })
             }
-            _ => Err("Unsupported literal expression.".to_owned()),
+            // Unary(ExprUnary { attrs: [], op: Neg(Sub), expr: Lit(ExprLit { attrs: [], lit: Int(LitInt { token: Literal { lit: 1 } }) }) }))
+            syn::Expr::Unary(syn::ExprUnary {
+                attrs: _,
+                ref op,
+                ref expr,
+            }) => match *op {
+                UnOp::Neg(_) => {
```

Comment:
While at it, maybe worth adding `!` as well? Seems easy enough to add another branch.

Reply:
Well, I tried to keep it simple as I did this in my free time on a week end.
Negative integer constants seem to be common while `pub const FALSE: bool = !true;` looks silly.
I would wait for somebody to have this problem.
There are other things I would like cbindgen being able to handle like 
`WARN: Skip constant::BAR - (Unhandled const definition)` from 
`const BAR: &'static str = "hello world";` which is very common, I think.

Reply:
Fair enough :) Sorry for derailing the PR.